automation/check-patch.e2e-kubemacpool-functests.sh

```diff
@@ -28,9 +28,27 @@ main() {
     echo "Deploy KubeVirt latest stable release"
     ./hack/deploy-kubevirt.sh
 
+    echo "Get skip tests regex"
+    SKIPPED_TESTS=$(__get_skipped_tests)
+
     # Run KubeMacPool functional tests
     cd ${TMP_COMPONENT_PATH}
-    KUBECONFIG=${KUBECONFIG} E2E_TEST_ARGS="--junit-output=$ARTIFACTS/junit.functest.xml" make functest
+
+    KUBECONFIG=${KUBECONFIG} E2E_TEST_ARGS="--junit-output=$ARTIFACTS/junit.functest.xml --ginkgo.skip $SKIPPED_TESTS" make functest
+}
+
+function __get_skipped_tests() {
+	local skipped_regex=""
+
+	if [ ! -z "$(grep 'default/mutatevirtualmachines_opt_in_patch.yaml' hack/components/bump-kubemacpool.sh)" ]; then
+		echo "KMP VM webhook is set to opt-out mode. skipping opt-in Context"
+		skipped_regex="${skipped_regex} \(opt-in\smode\)"
+	elif [ ! -z "$(grep 'default/mutatevirtualmachines_opt_in_patch.yaml' hack/components/bump-kubemacpool.sh)" ]; then
+		echo "KMP VM webhook is set to opt-in mode. skipping opt-out Context"
+		skipped_regex="${skipped_regex} \(opt-out\smode\)"
+	fi
+
+	echo $skipped_regex
```

Comment:
the function return its value via print, but you have more prints, 
how come it will work ?
i will suggest to return a value by setting a global for example


another point that will do problem is that if someone add debug prints in the function, it will change the logic and the return value in its current state

Reply:
ok. tell me what you think of [this change](https://github.com/kubevirt/cluster-network-addons-operator/compare/266058b750e06c72c3a4cf48b64496aacb8cba05..c9034d2d0005f3c0a2955a6f92251232a9b5d8f0) then

Reply:
i would prefer to keep it simple please
eval is $%%^
looks like we looked on the same site https://www.linuxjournal.com/content/return-values-bash-functions

unless you insist not to use global i would suggest this

```
function myfunc()
{
    myresult='some value'
}

myfunc
echo $myresult
```

Reply:
I don't want to use globals unless necessary. 
what non-global option would you suggest?

Reply:
btw i dont think on bash is such a bad idea,

anyhow i would suggest to remove the other echos, or if you want to echo them just make the function body non function and just embed it in the main file imo

Reply:
I'm good with using eval. I want to keep this generic function, and seems like eval makes the best out of all worlds.